pkg/util/syncutil/set.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package syncutil
+
+// Set is like a Go map[V]struct{} but is safe for concurrent use by multiple
+// goroutines without additional locking or coordination.
+// Loads, stores, and deletes run in amortized constant time.
+//
+// See the Map type for more details.
+type Set[V comparable] struct {
+	m Map[V, struct{}]
+}
+
+// Contains returns whether the value is stored in the set.
+func (s *Set[V]) Contains(value V) bool {
```

Comment:
nit: Consider adding a note saying using `Contains` on a `syncutil.Map` should only be done by tests. Calls to `Contains` before `Remove` are not generally safe to make, instead use the return value of `Remove`

Reply:
Good call. I added a comment discouraging use of `Contains` in code that also calls `Add` or `Remove`.